packages/components/src/ProgressBar/ProgressBar.tsx

```diff
@@ -19,15 +19,53 @@ interface ProgressBarProps {
    * @default base
    */
   readonly size?: keyof typeof sizes;
+
+  /**
+   * Set the variation of the progress bar
+   * @default progress
+   */
+  readonly variation?: "progress" | "stepped";
 }
 
 export function ProgressBar({
   currentStep,
   totalSteps,
   size = "base",
+  variation = "progress",
 }: ProgressBarProps) {
   const percentage = (currentStep / totalSteps) * 100;
   const progressBarClassName = classnames(styles.ProgressBar, sizes[size]);
+  const steppedProgressBarClassName = classnames(
+    styles.SteppedProgressBar,
+    sizes[size],
+  );
+
+  function steppedVariation() {
+    return (
+      <div className={steppedProgressBarClassName}>
+        {Array.from({ length: totalSteps }).map((_, index) => {
+          const step = index + 1;
+          const value = step <= currentStep ? 100 : 0;
+          const ariaValue = currentStep / totalSteps;
+          const ariaLabel = `progress, ${step} out of ${totalSteps}`;
+
+          return (
+            <progress
```

Comment:
to chris' point about voiceover, this is gonna get read out separately and since this is a non-standard way of showing progress, i'd use a `progressbar` role https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role along with the other `aria`s to read out the correct information.